website/project/model.py

```diff
@@ -3039,3 +3034,116 @@ def approve_embargo(self, user, token):
+
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    registration_metadata = fields.DictionaryField({})
+    registration_schema = fields.ForeignField('metaschema')
+    registered_node = fields.ForeignField('node')
+
+    storage = fields.ForeignField('osfstoragenodesettings')
+
+    config = fields.DictionaryField()
+    flags = fields.DictionaryField()
+
+    def __init__(self, *args, **kwargs):
+        super(DraftRegistration, self).__init__(*args, **kwargs)
+
+        meta_schema = self.registration_schema or kwargs.get('registration_schema')
+        if meta_schema:
+            schema = meta_schema.schema
+            config = schema.get('config', {})
+            self.config = config
+            flags = schema.get('flags', {})
+            for flag, value in flags.iteritems():
+                self.flags[flag] = value
+
+    # proxy fields from branched_from Node
+    def __getattr__(self, attr):
+        try:
+            return self.__dict__[attr]
+        except KeyError:
+            return getattr(self.branched_from, attr, None)
+
+    def find_question(self, qid):
+        for page in self.registration_schema.schema['pages']:
+            for question_id, question in page['questions'].iteritems():
+                if question_id == qid and 'description' in question:
+                    return question['description']
+
+    def get_comments(self):
+        """ Returns a list of all comments made on a draft in the format of :
+        [{
+          [QUESTION_ID]: {
+            'question': [QUESTION],
+            'comments': [LIST_OF_COMMENTS]
+            }
+        },]
+        """
+
+        all_comments = []
+        for question_id, value in self.registration_metadata.iteritems():
+            all_comments.append({
+                question_id: {
+                    'question': self.find_question(question_id),
+                    'comments': value['comments'] if 'comments' in value else ''
+                }
+            })
+        return all_comments
+
+    def get_flat_comments(self):
+        """ Returns a flat list of all comments made on a draft
+        """
+        flat_comments = list()
+        for question_id, value in self.registration_metadata.iteritems():
+            if 'comments' in value:
+                for comment in value['comments']:
+                    flat_comments.append(comment)
+        return flat_comments
+
+    def get_new_comments(self):
+        """ Returns a list of all comments admins haven't seen
+        In the same format as get_comments
+        """
+        comments = self.get_comments()
+
+        if comments:
+            for question_id, value in comments.iteritems():
+                for comment in value['comments']:
+                    if comment['adminHasSeen'] is False:
+                       comments.remove(question_id)
+
+        return comments
+
+    def has_new_comments(self):
```

Comment:
@barbour-em 
Now that we're not using adminHasSeen, lets either delete or update these.